js/5-todo-api/my-todo/src/todos.js

```diff
@@ -127,23 +128,23 @@ app.post(
       "date": "2018-08-25"
     }
 */
+
 app.put(
   '/todos/:id',
   decorateStoredTodosMiddleware,
   decorateStoredTodo,
   (req, res) => {
     const { todos } = res.locals
-    const updatedTodo = req.body
-
-    if (req.todo.id !== updatedTodo.id) {
-      res.status(400).send('Cannot update Todo Id')
-    } else {
-      todos[req.todoIndex] = updatedTodo
-      saveTodos(todos)
-      res.status(201).send(todos[req.todoIndex])
-    }
+    const storedTodo = res.locals.todo
+    const todoInput = req.body
+    const updatedTodo = { ...storedTodo, ...todoInput }
```

Comment:
This should be done in patch instead. Here you could use todoInput as the updatedTodo